docs/design/features/byreflike-generics.md

```diff
@@ -0,0 +1,75 @@
+# Generics parameters of ByRefLike types
+
+Using ByRefLike types in Generic parameters is possible by building upon support added for `ref` fields. Scenarios that would benefit most from this are those involving `Span<T>`. For example, consider the following examples:
+
+- `Span<TypedReference>` &ndash; Represents the general case where a ByRefLike type is used as a Generic parameter. This specific case would be desirable for a more efficient Reflection API.
+- `Span<Span<char>>` &ndash; Nested `Span<T>` types would be of benefit in the parsing result of strings.
+
+## Runtime impact
+
+Supporting ByRefLike type as Generic parameters will impact the following IL instructions:
+
+- `box` &ndash; Types with ByRefLike parameters used in fields cannot be boxed.
+- `throw` &ndash; Requires an object reference on the stack so not directly impacted since boxing is not permitted.
+- `stsfld` / `ldsfld` &ndash; Type fields of a ByRefLike parameter cannot be marked `static`.
```

Comment:
Does this mean that the generic type can have static field like that, but all attempts to access it via IL instructions or via reflection are going to fail?

Reply:
Does it mean that this is going to be C# compile error?
```
class G<AnyT> where AnyT : any
{
    static AnyT s;
}
```

Reply:
> Does it mean that this is going to be C# compile error?

I don't believe so. This would only occur if `AnyT` was a ByRefLike type _and_ was accessed at run-time. Is the point here that `static` fields are automatically initialized during type load?

Reply:
It seems like it would be best to make it an error in C#.

Reply:
@RikkiGibson That is fine. My point was that the runtime will handle this during type instantiation. As far as what C# does that is up to the language, the runtime will handle the case either way.